zummon/src/commonTest/kotlin/companysurvey/QuarterValidationTest.kt

```diff
@@ -0,0 +1,162 @@
+            )
+        )
+        val result = electricityValidator.validateQuarterHourlyDeliveryData(electricity)
+        assertEquals(Status.VALID, result.status)
+        assertEquals("Quarter-hourly delivery data has no gaps exceeding the limit", result.message)
+    }
+
+    @Test
+    fun validateQuarterHourlyDeliveryDataGapsExceed() {
+        val electricity = Electricity(
+            quarterHourlyDelivery_kWh = TimeSeries(
+                type = TimeSeriesType.ELECTRICITY_DELIVERY,
+                start = Instant.parse("2022-01-01T00:00:00Z"),
+                values = FloatArray(385) { 0.0f } + floatArrayOf(1.2f, 2.2f) // 385 nulls, exceeding limit
+            )
+        )
+        val result = electricityValidator.validateQuarterHourlyDeliveryData(electricity)
+        assertEquals(Status.INVALID, result.status)
+        assertEquals("Quarter-hourly delivery data has a gap of 96 hours, exceeding the allowed limit", result.message)
+    }
+
+    @Test
+    fun validateQuarterHourlyProductionDataInvalidGaps() {
+        val electricity = Electricity(
+            quarterHourlyProduction_kWh = TimeSeries(
+                type = TimeSeriesType.ELECTRICITY_PRODUCTION,
+                start = Instant.parse("2022-01-01T00:00:00Z"),
+                values = FloatArray(385) { 0.0f } + floatArrayOf(1.2f, 2.2f)// 385 nulls, exceeding limit
+            )
+        )
+        val result = electricityValidator.validateQuarterHourlyProductionData(electricity)
+        assertEquals(Status.INVALID, result.status)
+        assertEquals("Quarter-hourly production data has a gap of 96 hours, exceeding the allowed limit", result.message)
+    }
+
+    @Test
+    fun validateQuarterHourlyDeliveryDataExactLimit() {
+        val electricity = Electricity(
+            quarterHourlyDelivery_kWh = TimeSeries(
+                type = TimeSeriesType.ELECTRICITY_DELIVERY,
+                start = Instant.parse("2022-01-01T00:00:00Z"),
+                values = FloatArray(384) { 0.0f } + floatArrayOf(1.2f, 2.2f) // 384 nulls, at the limit
+            )
+        )
+        val result = electricityValidator.validateQuarterHourlyDeliveryData(electricity)
+        assertEquals(Status.VALID, result.status)
+        assertEquals("Quarter-hourly delivery data has no gaps exceeding the limit", result.message)
+    }
+
+    @Test
+    fun testQuarterHourlyFeedInLowProductionBatteryPower_ValidData() {
+        // Set up mock data with valid feed-in, production, and battery power
+        val gridConnection = GridConnection(
+            electricity = Electricity(
+                quarterHourlyFeedIn_kWh = TimeSeries(type= TimeSeriesType.ELECTRICITY_DELIVERY, start = Instant.parse("2024-01-01T00:00:00Z"), values = floatArrayOf(1.0f, 1.5f, 2.0f)),
+                quarterHourlyProduction_kWh = TimeSeries(type= TimeSeriesType.ELECTRICITY_DELIVERY, start = Instant.parse("2024-01-01T00:00:00Z"), values = floatArrayOf(1.0f, 1.5f, 2.0f))
```

Comment:
Nit: I think it would be a better test it the production values were slightly higer then feed-in